extensions/wikia/EditPageLayout/js/editor/AceEditor.js

```diff
@@ -0,0 +1,199 @@
+define('wikia.editpage.ace.editor', ['wikia.ace.editor'], function(ace){
+	'use strict';
+
+	var theme = 'solarized_light',
+		isEditorWide = false,
+		inputAttr = {
+			name: 'wpTextbox1',
+			id: 'wpTextbox1'
+		},
+		$editPage = $('#EditPage'),
+		$editPageToolbar = $('#EditPageToolbar'),
+		wideClassName = 'editpage-sourcewidemode-on',
+		narrowClassName = 'editpage-sourcewidemode-off',
+		beforeEditorInit = function(){
+			$('.loading-indicator').remove();
+			$('#wpSave').removeAttr( 'disabled' );
+			$editPage.addClass('editpage-sourcewidemode mode-source ' + narrowClassName);
+		};
+
+	function init() {
+		chooseTheme();
+		initConfig();
+		beforeEditorInit();
+		ace.init( 'editarea', inputAttr );
+		initOptions();
+		initStyles();
+		ace.setTheme( theme );
+		ace.setMode( window.codePageType );
+
+		initEvents();
+	}
+
+	/**
+	 * Init ace editor config
+	 */
+	function initConfig() {
+		var config = {
+				workerPath: window.aceScriptsPath
+			};
+
+		ace.setConfig( config );
+	}
+
+	/**
+	 * Init ace editor options
+	 */
+	function initOptions() {
+		var options = {
+			showPrintMargin: false
+		};
+
+		ace.setOptions( options );
+	}
+
+	/**
+	 * Init ace editor custom styles
+	 */
+	function initStyles() {
+		var editor = ace.getEditor();
+
+		editor.css('fontSize', '14px');
+	}
+
+	/**
+	 * Init submit action
+	 */
+	function initSubmit() {
+		$( '#editform' ).submit(function(e) {
+			var $form = $( this ),
```

Comment:
You could use 
$(this)
.append(hiddenInput)
.submit();
so there's no need of using a form var here :)

Also, I guess preventing default action could be moved to the top of the handler to more easily indicate that there are additional steps done before resubmitting :dancer: